templates/remix/app/entry.server.tsx

```diff
@@ -14,6 +14,17 @@ export default function handleRequest(
 
   responseHeaders.set("Content-Type", "text/html");
 
+  // Add Link header for HTTP/2 Server Push
+  let http2PushLinksHeaders = remixContext.matches.reduce((acc: string[], { route: { module, imports } }) => {
+    acc.push(module);
+    if (imports) {
+      acc.push(...imports);
+    }
+    return acc;
+  }, []);
+  http2PushLinksHeaders.push(...[remixContext.manifest.url, remixContext.manifest.entry.module, ...remixContext.manifest.entry.imports])
```

Comment:
This could be simplified to

```suggestion
  let http2PushLinksHeaders = remixContext.matches
    .flatMap(({ route: { module, imports } }) => [module, ...(imports || [])])
    .filter(Boolean)
    .concat([
      remixContext.manifest.url,
      remixContext.manifest.entry.module,
      ...remixContext.manifest.entry.imports,
    ]);
```